internal/v1/api.yaml

```diff
@@ -571,11 +572,21 @@ paths:
           example: '123e4567-e89b-12d3-a456-426655440000'
           required: true
           description: UUID of a blueprint
-      summary: create new compose from blueprint
-      description: "create new compose from blueprint"
+      summary: create new compose from blueprint, Users can optionally specify the target image types to be built for this composition.
+      description: "create new compose from blueprint, optionally specifying the target image types to build"
       operationId: composeBlueprint
       tags:
         - blueprint
+      requestBody:
+          required: false
+          description: "list of target image types that the user wants to build for this compose"
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: "#/components/schemas/ImageTypes"
+                  example: ["azure", "aws"]
```

Comment:
Please update the schema so the body looks like this:

```json
{
 "types": [ "..." ]
}
```

Not like this:

```json
[ "..." ]
```

Make sure the following input means "build me all types". Example 1 - an empty body. Example 2 - a body that has an empty JSON:

```json
{ }
```

In contrast, the following example means "build me nothing" not "build me all types from blueprint":

```json
{
 "types": [ ]
}
```

At least this is how I see the most logical semantics for the endpoint. What others think?